mypy/checkmember.py

```diff
@@ -91,10 +92,24 @@ def analyze_member_access(name: str, typ: Type, node: Context, is_lvalue: bool,
         if isinstance(ret_type, TupleType):
             ret_type = ret_type.fallback
         if isinstance(ret_type, Instance):
-            result = analyze_class_attribute_access(ret_type, name, node, is_lvalue,
-                                                    builtin_type, not_ready_callback, msg)
-            if result:
-                return result
+            if not isinstance(node, (OpExpr, ComparisonExpr)):
```

Comment:
I'd consider adding an optional bool parameter to this function instead and setting it in `check_op` (in `checkexpr.py`).